diff --git a/scripts/core_common/modules/boost.py b/scripts/core_common/modules/boost.py
index af003e7..e6b9d99 100644
--- a/scripts/core_common/modules/boost.py
+++ b/scripts/core_common/modules/boost.py
@@ -65,0 +66 @@ def make():
+  base.replaceInFile("./tools/build/src/tools/msvc.jam", "limit = \"-version \\\"[16.0,17.0)\\\"\" ;", "limit = \"-version \\\"[16.0,18.0)\\\"\" ;")
@@ -75,0 +77,3 @@ def make():
+    win_toolset = os.environ.get("EO_WINDOWS_BOOST_TOOLSET", win_toolset)
+    win_boot_arg = os.environ.get("EO_WINDOWS_BOOST_BOOTSTRAP", win_boot_arg)
+    win_vs_version = os.environ.get("EO_WINDOWS_BOOST_LIB_TAG", win_vs_version)
@@ -80,3 +84,2 @@ def make():
-      base.cmd("b2.exe", ["headers"])
-      base.cmd("b2.exe", ["--clean"])
-      base.cmd("b2.exe", ["--prefix=./../build/win_64", "link=static", "--with-filesystem", "--with-system", "--with-date_time", "--with-regex", "--toolset=" + win_toolset, "address-model=64", "install"])
+      base.cmd2("b2.exe", ["headers"])
+      base.cmd2("b2.exe", ["--prefix=./../build/win_64", "--stagedir=./../build/win_64/stage", "--build-dir=./../build/win_64/build-temp", "--layout=tagged", "variant=release", "architecture=x86", "address-model=64", "binary-format=pe", "abi=ms", "link=static", "--with-filesystem", "--with-system", "--with-date_time", "--with-regex", "--toolset=" + win_toolset, "install"])
@@ -85,3 +88,2 @@ def make():
-      base.cmd("b2.exe", ["headers"])
-      base.cmd("b2.exe", ["--clean"])
-      base.cmd("b2.exe", ["--prefix=./../build/win_32", "link=static", "--with-filesystem", "--with-system", "--with-date_time", "--with-regex", "--toolset=" + win_toolset, "address-model=32", "install"])
+      base.cmd2("b2.exe", ["headers"])
+      base.cmd2("b2.exe", ["--prefix=./../build/win_32", "--stagedir=./../build/win_32/stage", "--build-dir=./../build/win_32/build-temp", "--layout=tagged", "variant=release", "architecture=x86", "address-model=32", "binary-format=pe", "abi=ms", "link=static", "--with-filesystem", "--with-system", "--with-date_time", "--with-regex", "--toolset=" + win_toolset, "install"])
