diff --git a/scripts/core_common/modules/boost.py b/scripts/core_common/modules/boost.py
index 9458064..4b8723f 100644
--- a/scripts/core_common/modules/boost.py
+++ b/scripts/core_common/modules/boost.py
@@ -79,12 +79,12 @@ def make():
       base.cmd("bootstrap.bat", [win_boot_arg])
       base.cmd("b2.exe", ["headers"])
       base.cmd("b2.exe", ["--clean"])
-      base.cmd("b2.exe", ["--prefix=./../build/win_64", "link=static", "--with-filesystem", "--with-system", "--with-date_time", "--with-regex", "--toolset=" + win_toolset, "address-model=64", "install"])
+      base.cmd("b2.exe", ["--prefix=./../build/win_64", "--stagedir=./../build/win_64/stage", "--build-dir=./../build/win_64/build-temp", "--layout=tagged", "variant=release", "architecture=x86", "address-model=64", "binary-format=pe", "abi=ms", "link=static", "--with-filesystem", "--with-system", "--with-date_time", "--with-regex", "--toolset=" + win_toolset, "install"])
     if (-1 != config.option("platform").find("win_32")) and not base.is_file("../build/win_32/lib/libboost_system-" + win_vs_version + "-mt-x32-1_72.lib"):
       base.cmd("bootstrap.bat", [win_boot_arg])
       base.cmd("b2.exe", ["headers"])
       base.cmd("b2.exe", ["--clean"])
-      base.cmd("b2.exe", ["--prefix=./../build/win_32", "link=static", "--with-filesystem", "--with-system", "--with-date_time", "--with-regex", "--toolset=" + win_toolset, "address-model=32", "install"])
+      base.cmd("b2.exe", ["--prefix=./../build/win_32", "--stagedir=./../build/win_32/stage", "--build-dir=./../build/win_32/build-temp", "--layout=tagged", "variant=release", "architecture=x86", "address-model=32", "binary-format=pe", "abi=ms", "link=static", "--with-filesystem", "--with-system", "--with-date_time", "--with-regex", "--toolset=" + win_toolset, "install"])
     if (-1 != config.option("platform").find("win_arm64") and not base.is_file("../build/win_arm64/lib/libboost_system-" + win_vs_version + "-mt-a64-1_72.lib")):
       boost_bat = []
       boost_bat.append("call bootstrap.bat " + win_boot_arg) # first build b2 for win64, so vcvarsall_call with arm64 later
